sdk/app_cpu1/common/sys/cmd/cmd_hw.c

```diff
@@ -246,19 +233,61 @@ int cmd_hw(int argc, char **argv)
         }
     }
 
-    // Handle 'timer' sub-command
-    if (argc >= 2 && STREQ("timer", argv[1])) {
-        if (argc == 4 && STREQ("fpga", argv[2]) && STREQ("now", argv[3])) {
-            uint32_t counts1 = fpga_timer_now();
-            uint32_t counts2 = fpga_timer_now();
+    // Handle 'timing manager' sub-command
+    if (argc >= 2 && STREQ("timing", argv[1]) && STREQ("manager", argv[2])) {
+        // hw timing manager trigger <HIGH|LOW|BOTH>
+        if (argc == 5 && STREQ("trigger", argv[3])) {
+            if (STREQ("HIGH", argv[4])) {
+                timing_manager_trigger_on_pwm_high();
+            } else if (STREQ("LOW", argv[4])) {
+                timing_manager_trigger_on_pwm_low();
+            } else if (STREQ("BOTH", argv[4]))
+                timing_manager_trigger_on_pwm_both();
+            else {
+                return CMD_INVALID_ARGUMENTS;
+            }
+            return CMD_SUCCESS;
+        }
 
-            cmd_resp_printf("counts1: %lu\r\n", counts1);
-            cmd_resp_printf("counts2: %lu\r\n", counts2);
-            cmd_resp_printf("time delta = %8.3f ns\r\n", 1e3 * fpga_timer_ticks_to_usec(counts2 - counts1));
+        // hw timing manager ratio <count>
+        else if (argc == 5 && STREQ("ratio", argv[3])) {
+            uint32_t ratio = (uint32_t)(atoi(argv[4]));
+            timing_manager_set_ratio(ratio);
+            return CMD_SUCCESS;
+        }
 
+        // "hw timing manager enable <sensor> <port [if eddy]>
+        else if (argc >= 5 && STREQ("enable", argv[3])) {
+            if (STREQ("encoder", argv[4])) {
+                timing_manager_enable_encoder();
+            } else if (STREQ("adc", argv[4])) {
+                timing_manager_enable_adc();
+            } else if (argc == 6 && STREQ("eddy", argv[4])) {
+                int32_t port = atoi(argv[5]);
+                // enable eddy current sensor based on selected port
+                if (port == 1) {
+                    timing_manager_enable_eddy_1();
+                } else if (port == 2) {
+                    timing_manager_enable_eddy_2();
+                } else if (port == 3) {
+                    timing_manager_enable_eddy_3();
+                } else if (port == 4) {
+                    timing_manager_enable_eddy_4();
+                }
+            }
             return CMD_SUCCESS;
         }
 
+        // 'hw timing manager <sensor> time'
+        else if (argc >= 5 && STREQ("time", argv[4])) {
+            if (STREQ("encoder", argv[3])) {
+                timing_manager_get_time_per_sensor(ENCODER);
+            }
+        }
+    }
```

Comment:
Seems to not support the other sensors?

Reply:
have since added other sensors, including in this commit